tests/unit/test_http.py

```diff
@@ -102,31 +109,48 @@ async def test_session_stream(mock_request):
 
 @respx.mock
 @pytest.mark.asyncio
-async def test_session_request_retry(mock_request, mock_response):
+async def test_session_request_retry(monkeypatch, mock_request):
+    """Test the retry in the Session.request method"""
     async with http.Session() as ps:
         route = respx.get(TEST_URL)
         route.side_effect = [
             httpx.Response(HTTPStatus.TOO_MANY_REQUESTS, json={}),
             httpx.Response(HTTPStatus.OK, json={})
         ]
 
-        ps.retry_wait_time = 0  # lets not slow down tests for this
+        # let's not actually introcude a wait into the tests
+        monkeypatch.setattr(http, 'MAX_RETRY_BACKOFF', 0)
         resp = await ps.request(mock_request)
         assert resp
         assert route.call_count == 2
 
 
 @respx.mock
 @pytest.mark.asyncio
-async def test_session_retry(mock_request):
-    async with http.Session() as ps:
+async def test_session__retry(monkeypatch):
+    """A unit test for the _retry function"""
+    # decorator form of patch seems to conflict with respx.mock
```

Comment:
@jreiberkyle it looks to me like this is intended behavior for patch. Under https://docs.python.org/3/library/unittest.mock.html#unittest.mock.patch it says

> If [patch()](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.patch) is used as a decorator and new is omitted, the created mock is passed in as an extra argument to the decorated function.

Someone opened a ticket about this at https://bugs.python.org/issue44052, but they're looking at it from the other side: they don't want to add a new argument when they use `new`.

Reply:
That sleep gets called with a particular value is kind of an implementation detail, tho? Often we want to avoid testing these because they get in the way of refactoring down the road. The backoff timing logic could be checked by testing _calc_exponential_wait directly, yes?

Here's an idea: your test_func could assert that requests were backing off by recording some timestamps. That's not an implementation detail and sorta mimics what we're trying to do for the server?

Reply:
I didn't observe any conflict with respx.mock. What I saw was that patch works but doesn't pass the mock to the function if the new keyword argument is used. And then pytest mistakes the argument for a fixture and fails.

Reply:
Ok, well it seems we are stuck with this way for mocking an async function since we support pre-3.8 and only at 3.8 does it return an AsyncMock by default.

Reply:
@jreiberkyle here's an example of using unittest.mock.patch and respx.mock together. I'm patching OrderStates in test_checksum_success and printing the value of OrderStates in orders.py to show that it's a MagicMock. The test passes and I see "MagicMock" printed, so apparently unittest.mock.patch and respx.mock are both working.

```diff
diff --git a/planet/clients/orders.py b/planet/clients/orders.py
index 6b8c7c8..bcb906f 100644
--- a/planet/clients/orders.py
+++ b/planet/clients/orders.py
@@ -348,6 +348,7 @@ class OrdersClient:
         """
         order = await self.get_order(order_id)
         order_state = order['state']
+        print(OrderStates)
         if not OrderStates.is_final(order_state):
             raise exceptions.ClientError(
                 'Order cannot be downloaded because the order state '
diff --git a/tests/integration/test_orders_api.py b/tests/integration/test_orders_api.py
index bf8bd16..108dcb9 100644
--- a/tests/integration/test_orders_api.py
+++ b/tests/integration/test_orders_api.py
@@ -19,10 +19,12 @@ import math
 import os
 from pathlib import Path
 from unittest.mock import call, create_autospec
+from unittest import mock
 import hashlib
 
 import httpx
 import pytest
+# from requests import patch
 import respx
 
 from planet import OrdersClient, exceptions, reporting
@@ -551,7 +553,8 @@ async def test_download_asset_md(tmpdir, session):
 @respx.mock
 @pytest.mark.asyncio
 @pytest.mark.parametrize("checksum", [("MD5"), ("SHA256")])
-async def test_checksum_success(tmpdir,
+@mock.patch("planet.clients.orders.OrderStates", autospec=True)
+async def test_checksum_success(mock_states, tmpdir,
                                 order_description,
                                 oid,
                                 session,
```